be/src/exec/vectorized/hdfs_scanner_text.h

```diff
@@ -0,0 +1,118 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "exec/vectorized/hdfs_scanner.h"
+#include "formats/csv/converter.h"
+
+namespace starrocks::vectorized {
+
+class HdfsTextScanner final : public HdfsScanner {
+#ifndef BE_TEST
+    constexpr static size_t kMinBufferSize = 8 * 1024 * 1024L;
+    constexpr static size_t kMaxBufferSize = 512 * 1024 * 1024L;
+#else
+    constexpr static size_t kMinBufferSize = 128 * 1024L;
+    constexpr static size_t kMaxBufferSize = 512 * 1024L;
+#endif
+
+public:
+    HdfsTextScanner() = default;
+    ~HdfsTextScanner() override = default;
+
+    Status do_open(RuntimeState* runtime_state) override;
+    void do_close(RuntimeState* runtime_state) noexcept override;
+    Status do_get_next(RuntimeState* runtime_state, ChunkPtr* chunk) override;
+    Status do_init(RuntimeState* runtime_state, const HdfsScannerParams& scanner_params) override;
+    Status _parse_csv(ChunkPtr* chunk);
+
+private:
+    class Buffer {
+    public:
+        // Does NOT take the ownership of |buff|.
+        Buffer(char* buff, size_t cap) : _begin(buff), _position(buff), _limit(buff), _end(buff + cap) {}
+
+        void append(char c) { *_limit++ = c; }
+
+        // Returns the number of bytes between the current position and the limit.
+        size_t available() const { return _limit - _position; }
+
+        // Returns the number of elements between the the limit and the end.
+        size_t free_space() const { return _end - _limit; }
+
+        // Returns this buffer's capacity.
+        size_t capacity() const { return _end - _begin; }
+
+        // Returns this buffer's read position.
+        char* position() { return _position; }
+
+        // Returns this buffer's write position.
+        char* limit() { return _limit; }
+
+        void add_limit(size_t n) { _limit += n; }
+
+        // Finds the first character equal to the given character |c|. Search begins at |pos|.
+        // Return: address of the first character of the found character or NULL if no such
+        // character is found.
+        char* find(char c, size_t pos = 0) { return (char*)memchr(position() + pos, c, available() - pos); }
+
+        void skip(size_t n) { _position += n; }
+
+        // Compacts this buffer.
+        // The bytes between the buffer's current position and its limit, if any,
+        // are copied to the beginning of the buffer.
+        void compact() {
+            size_t n = available();
+            memmove(_begin, _position, available());
+            _limit = _begin + n;
+            _position = _begin;
+        }
+
+    private:
+        char* _begin;
+        char* _position; // next read position
+        char* _limit;    // next write position
+        char* _end;
+    };
+
+    class CSVReader {
```

Comment:
AFAIK, there is also a class `CSVReader` in csv_scaner.cpp. Is it possible to write `CSVReader` class shared by hdfs_scanner and csv_scanner?

Reply:
there is a little difference between two `CSVReader`, i will do a refactor for this @dirtysalt thanks